### ISSUE TITLE
Could not find module `fetch` imported from `ember-data/adapters/rest` on a newly generated app with ember-cli-fastboot

### ISSUE THREAD

**Issue by mansona** [emberjs/data#5998]
### Reproduction

Steps to reproduce: 

- ember new test-app
- cd test-app
- ember install ember-cli-fastboot
- npm start
- visit http://localhost:4200 
- 💥 

here is a simple reproduction (because it was asked for) https://github.com/mansona/data-fastboot-test

### Description

I understand that there are things required for enabling dependencies in fastboot but this flow used to work and it should still work i.e. we shouldn't expect people to do extra steps to get fastboot working for something that is included in the default blueprint

### Versions

```cli
data-fastboot-test@0.0.0 /Users/mansona/temp/data-fastboot-test
└── ember-source@3.9.1

data-fastboot-test@0.0.0 /Users/mansona/temp/data-fastboot-test
└── ember-cli@3.9.0

data-fastboot-test@0.0.0 /Users/mansona/temp/data-fastboot-test
└── ember-data@3.9.0
```

**Comment by runspired** [comment]
@mansona I think this is an issue with one of the fastboot dependencies vs `ember-data`. We do ensure that `fetch` is present: https://github.com/emberjs/data/blob/v3.9.0/package.json#L57

I think someone noticed once before though (but having trouble finding the ticket) that fastboot is bringing it's own and clobbering something in a bad way.

**Comment by mansona** [comment]
@runspired does it have anything to do with https://github.com/ember-fastboot/ember-cli-fastboot/issues/588?

**Comment by rwjblue** [comment]
The issue is that ember-cli-fastboot only calls `updateFastbootManifest` on top level addons, ember-fetch needs to use that hook to provide the correct node `fetch` (vs the browser one).

@kiwiupover has been working on this for a bit but has hit a few yak shaves as he goes (the ember-cli-fastboot test suite itself is borked due to ember-data monorepo changes for example).

**Comment by runspired** [comment]
This was resolved in 3.9.1 and again in 3.9.3 (after 3.9.2 regressed it).
